packages/theme/modules/catalog/product/types.ts

```diff
@@ -0,0 +1,39 @@
+import {
+  BundleProduct as BaseBundleProduct,
+  ConfigurableProduct as BaseConfigurableProduct,
+  DownloadableProduct as BaseDownloadableProduct,
+  GroupedProduct as BaseGroupedProduct, ProductAttributeFilterInput, ProductAttributeSortInput,
+  ProductInterface,
+  VirtualProduct as BaseVirtualProduct,
+} from '~/modules/GraphQL/types';
+
+/**
+ * There is no __typename in GraphQL definitions but type_id is marked as a deprecated
+ */
+export interface Product extends ProductInterface, ConfigurableProduct, Omit<BundleProduct, 'items'>, Omit<GroupedProduct, 'items'>, Omit<DownloadableProduct, 'items'>, Omit<VirtualProduct, 'items'> { __typename: string }
+export interface BundleProduct extends BaseBundleProduct { __typename: string }
```

Comment:
```suggestion
export interface BundleProduct extends BaseBundleProduct & { __typename: string }
```

huh kinda surprised that this works without the `&`, if it *really* works then it's fine

Reply:
Outdated.